site/gatsby-site/migrations/2023.06.26T21.29.07.fix-gmf-multi.js

```diff
@@ -0,0 +1,90 @@
+const { isArray, isString } = require('lodash');
+
+const config = require('../config');
+
+/**
+ *
+ * @param {{context: {client: import('mongodb').MongoClient}}} context
+ */
+exports.up = async ({ context: { client } }) => {
+  await client.connect();
+
+  const fixValueJsonOptions = (value_json) => {
+    const value = JSON.parse(value_json);
+
+    const newValue = value
+      .map((v) => v.label)
+      .filter((v) => isString(v))
```

Comment:
Sometimes the array contains both objects (custom options) and strings (pre-existing ones). Mapping to the non-existent `v.label` and then filtering by `isString` removes some of the values. When I logged the old and new values, I got this:

```
GMF 470 Known AI Technical Failure Snippets
[
  {
    "attributes": [
      {
        "short_name": "Snippet Text",
        "value_json": "\"But just a few minutes of exploration by TechCrunch produced not just hateful rhetoric “in the style of Hitler,” but it repeated the same pandemic-related untruths noted by NewsGuard. As in it literally repeated them as the answer and cited ChatGPT’s generated disinfo (clearly marked as such in the original and in a NYT write-up) as the source.\""
      },
      {
        "short_name": "Related Classifications",
        "value_json": "[\"Problematic Input\",{\"customOption\":true,\"label\":\"Inadequate Verification\",\"id\":\"new-id-2\"}]"
                          ^^^^^^^^^^^^^^^^^

      }
    ]
  }
]
New:
[
  {
    "attributes": [
      {
        "short_name": "Snippet Text",
        "value_json": "\"But just a few minutes of exploration by TechCrunch produced not just hateful rhetoric “in the style of Hitler,” but it repeated the same pandemic-related untruths noted by NewsGuard. As in it literally repeated them as the answer and cited ChatGPT’s generated disinfo (clearly marked as such in the original and in a NYT write-up) as the source.\""
      },
      {
        "short_name": "Related Classifications",
        "value_json": "[\"Inadequate Verification\"]"
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


      }
    ]
  }
]
```

Reply:
Good catch. I added a test for peace of mind.

I wonder what caused this to change from storing strings to objects 🤔 Having more tests would have probably caught that.

(I was going to create a new issue, but https://github.com/responsible-ai-collaborative/aiid/issues/1900 already exists)